src/plugins/zenuml.ts

```diff
@@ -0,0 +1,63 @@
+import { visit } from 'unist-util-visit';
+
+function attr(name, value) {
+  return {
+    type: 'mdxJsxAttribute',
+    name,
+    value,
+  };
+}
+
+async function handleNode(node) {
+  const code = encodeURIComponent(
+    JSON.stringify({
+      js: node.value,
+    }),
+  );
+  const params = new URLSearchParams(node.meta);
+
+  const jsxNode = {
+    type: 'mdxJsxTextElement',
+    name: 'div',
+    attributes: [attr('className', 'zenuml-embed')],
+    children: [
+      {
+        type: 'mdxJsxTextElement',
+        name: 'iframe',
+        attributes: [
+          attr(
+            'src',
+            `https://app.zenuml.com?embed=1&code=${code}&title=${
+              params.get('title') || ''
+            }&stickyOffset=30`,
+          ),
+          attr('width', '100%'),
+          attr('height', '600px'),
+          attr('frameBorder', '0'),
+        ],
+        children: [],
+      },
+    ],
+  };
+
+  Object.keys(node).forEach((key) => delete node[key]);
```

Comment:
Why does it need to iterate the keys? If the node is not going to mutate, can we simply assign jsxNode to node?
Or use `Object.assign({}, node)?

Reply:
This code is copied from [facebook/react-native-website](https://github.com/facebook/react-native-website/blob/b8cd3b012ad166e2cd38ef2cf0e7171ee46e0994/plugins/remark-snackplayer/src/index.js#L79). It's clear that the object reference must be preserved since the function doesn't return any value. I'm uncertain whether it's necessary to delete all keys - perhaps leaving some unwanted keys that affect the rendering?
Therefore, `node = jsxNode` or `Object.assign({}, node)` aren't viable options. However, `Object.assign(node, jsxNode)` could be a potential consideration.